### PR TITLE
Reduce Application Binary Size

### PR DESCRIPTION
**Pull request by jkmassel**
In #22296, we merged a very large PDF file as an `xcassets` resource.

This increased the binary size by 43.1MB, causing it to be too large for installation or update on cellular networks.

This PR resolves that by:

1. Switching out the PDF file for a JPEG-compressed gradient. If I'd had more time, I would've probably generated the gradient in code instead of as a static image, but that can be left for another day.
2. Running perceptually invisible compression on all PNG assets (which saves another 0.5 MB)

Screenshot 2024-05-27 at 3 18 52 PM

To test:

## Regression Notes
1. Potential unintended areas of impact
- Modifies many icon images – use the left/right comparison on GitHub to note that they're visually indistinguishable.
- Run the first-launch process, note that there's no user-visible difference.

3. What I did to test those areas of impact (or what existing automated tests I relied on)
- I did the items above.

4. What automated tests I added (or what prevented me from doing so)
- No automated tests available for this.
- 
PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/wordpress-mobile/WordPress-iOS/milestone/291">25.0 ❄️</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WordPress/Resources/AppImages.xcassets/stats-revamp-v2-feature-intro-preview.imageset/Contents.json
Comment (guarani) [inline WordPress/Resources/AppImages.xcassets/stats-revamp-v2-feature-intro-preview.imageset/Contents.json:1]: @jkmassel, this resource is unused and can be removed entirely. 
We removed the screen that used it in https://github.com/wordpress-mobile/WordPress-iOS/pull/19917, but the PDF was overlooked.

**Review by guarani — COMMENTED** [review]
Thanks @jkmassel!

I realized the Stats PDF isn't used, so it can be removed. 

I tested the other change to the welcome screen and noticed it looks odd in dark mode:

| this branch | `trunk` |
| - | - |
|  |  |

**Comment by alpavanoglu** [comment]
Hey @jkmassel 
> This increased the binary size by 43.1MB

So that I have a better understanding and avoid this kind of thing in the future, how does that pdf file increase the binary size by 43.1 Megabytes when the file itself appear to be 5 Kilobytes? I understand the vector file will be converted but still the number is extremely high indeed. 

If you mean the gradient, shouldn't the trade there lower the size rather than increase it?

Screenshot 2024-05-28 at 16 21 49

File: WordPress/Resources/AppImages.xcassets/stats-revamp-v2-feature-intro-preview.imageset/Contents.json
Comment (jkmassel) [reply WordPress/Resources/AppImages.xcassets/stats-revamp-v2-feature-intro-preview.imageset/Contents.json:1]: Great! Addressed in ac030705854d26713fbceaa301726ee45f49d726

**Comment by jkmassel** [comment]
@alpavanoglu – I'm not sure I understand?

> how does that pdf file increase the binary size by 43.1 Megabytes when the file itself appear to be 5 Kilobytes

Where did you get 5kb from? The binary size is increased so much because the 7 MB PDF file is converted to a 3x, 2x, and 1x PNG by Xcode during compilation. The resulting `Assets.car` file is duplicated in the App Binary – it's also part of `WordPressShareExtension`, `WordPressDraftActionExtension`,  and`JetpackIntents`, so the cost has to be paid four times.

> If you mean the gradient, shouldn't the trade there lower the size rather than increase it?

Interesting – I suspected #22296 because removing this file reduced the binary size by the 43 MB and we got a notification from Apple on Feb 20 that the app had exceeded the download limit. Perhaps there is another factor at play?

**Comment by jkmassel** [comment]
Thanks @guarani – the transparency adds a bunch of complexity – the issue should now be resolved. Unfortunately the savings aren't quite as dramatic as with a JPEG but it's a good start. Brings down the Jetpack binary size from 269.8MB to 210.6MB, so we're much closer to the cellular limit now!

**Comment by alpavanoglu** [comment]
@jkmassel 

> Where did you get 5kb from?

That PR added 1 PDF and replaced the other. The pure addition is the small one I referred to there. And the replaced PDF is smaller in size so I reckoned overall I would be reducing the size and not increase it. 

> Perhaps there is another factor at play?

Perhaps. I wanted to clarify to avoid such things in the future.

**Comment by jkmassel** [comment]
Addressed in #23381

**Pull request closed without merging**